File: html-css/src/updateprofile.py
import tornado.web
import json
import base64

class Handler(tornado.web.RequestHandler):
    def post(self):
        J=json.loads(self.request.body)
        realName = J["realName"]
        dob = J["birthDate"]
        profilePic = J.get("profilePic")
        if profilePic:
            ppic = base64.b64decode(profilePic)
            print("WE GOT:",realName,dob,ppic[:20])
        else:
            print("no profile pic")
        resp={"ok": True}
        self.write( json.dumps(resp) )

File: html-css/html/updateprofile.js
"use strict";

function submit() {
    let rname = document.getElementById("rname").value;
    let dob = document.getElementById("dob").value;
    let email = document.getElementById("email").value;

    let J = {
        realName: rname,
        birthDate: dob,
        email: email
    };

    fetch("/profile",
        {
            method: "POST",
            body: JSON.stringify(J)
        }
    ).then((resp) => {
        resp.json().then((J) => {
            console.log("Server said:", J);
            saveImage();
            document.location.reload();
        });
    }).catch((err) => {
        console.log("Uh oh", err);
    });
}

function saveImage() {
    let picture = document.getElementById("ppic").files[0];
    if(picture) {
        let R = new FileReader();
        R.addEventListener("load", () => {
            let profilePic = btoa(R.result);

            localStorage.setItem("profileImage", profilePic);
        });

        R.readAsDataURL(picture);
    }
}

window.onload = function() {
    let profilePicData = localStorage.getItem("profileImage");
    if (profilePicData) {
        document.getElementById("profileImage").src = "data:image/octet-stream;base64," + profilePicData;
    }
    localStorage.removeItem("profileImage");
}